setup.py

```diff
@@ -100,7 +99,14 @@
         "Source Code": "https://github.com/Qiskit/qiskit-terra",
     },
     rust_extensions=[
-        RustExtension("qiskit._accelerate", "crates/accelerate/Cargo.toml", binding=Binding.PyO3)
+        RustExtension(
+            "qiskit._accelerate",
+            "crates/accelerate/Cargo.toml",
+            binding=Binding.PyO3,
+            # If RUST_DEBUG is set, force compiling in debug mode. Else, use the default behavior
+            # of whether it's an editable installation.
+            debug=True if os.getenv("RUST_DEBUG") == 1 else None,
+        )
```

Comment:
This code is also used in https://github.com/Qiskit/qiskit-terra/pull/9757. Inspired by a PyO3 snippet